File: android/src/main/java/com/poptacular/plugins/ump/UmpPlugin.java
package com.poptacular.plugins.ump;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Ump")
public class UmpPlugin extends Plugin {

    private Ump implementation = new Ump();

    @PluginMethod
    public void initialise(PluginCall call) {
        implementation.initialise(this.bridge.getActivity());
        call.resolve();
    }
}

File: ios/Plugin/UmpPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UmpPlugin)
public class UmpPlugin: CAPPlugin {
    private let implementation = Ump()

    @objc func initialise(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.initialise((self.bridge?.viewController)!)
        ])
    }

    @objc func show(_ call: CAPPluginCall) {
        let value = call.getBool("value") ?? false
        call.resolve([
            "value": implementation.show((self.bridge?.viewController)!, force: value)
        ])
    }
}
